course/page/text.py

```diff
@@ -373,33 +374,60 @@ def __init__(self, vctx, location, matcher_desc):
                 matcher_desc,
                 required_attrs=(
                     ("type", str),
-                    ("value", (int, float)),
+                    ("value", (int, float, str)),
                     ),
                 allowed_attrs=(
-                    ("rtol", (int, float)),
-                    ("atol", (int, float)),
+                    ("rtol", (int, float, str)),
+                    ("atol", (int, float, str)),
                     ),
                 )
 
+        def validate_attr(attr):
+
+            attr_value=self.matcher_desc.__getattribute__(attr)
+
+            try:
+                eval(attr_value)
+            except:
+                raise ValidationError(
+                        string_concat(
+                            "%(location)s: ",
+                            _("attribute '%(attr)s' "
+                              "should be an instances "
+                              "of 'int', 'float' "
+                              "or a caculable string"))
+                        % {
+                            'location': location,
+                            'attr': attr})
+
+        validate_attr("value")
+
+        if hasattr(self.matcher_desc, "atol"):
+            validate_attr("atol")
+
+        if hasattr(self.matcher_desc, "rtol"):
+            validate_attr("rtol")
+
+
     def validate(self, s):
         try:
-            float(s)
+            float(eval(s))
         except:
             tp, e, _ = sys.exc_info()
             raise forms.ValidationError("%(err_type)s: %(err_str)s"
                     % {"err_type": tp.__name__, "err_str": str(e)})
 
     def grade(self, s):
-        answer_float = float(s)
+        answer_float = float(eval(s))
```

Comment:
This is a nonstarter, unfortunately, because this allows arbitrary code execution based on input into a web form.